### ISSUE TITLE
bug: KMS "Perhaps the blob didn't come from LocalStack" error message when decrypting using public key generated by localstack

### ISSUE THREAD

**Issue by getglad** [localstack/localstack#11313]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Localstack can generate a KMS `RSA_4096` key and export the public key. This key material can encrypt a payload on the client side using RSA OAEP. However, the KMS `decrypt` operation fails with the below message even when passing in the key id.

> operation error KMS: Decrypt, https response error StatusCode: 400, RequestID: 9f9efecf-6e08-447d-aef0-6c49dff32ed1, InvalidCiphertextException: LocalStack is unable to deserialize the ciphertext blob. Perhaps the blob didn't come from LocalStack

### Expected Behavior

Given the use of a key pair Localstack owns, I would expect the `decrypt` operation should be possible.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    `localstack start` (homebrew install)

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

It fails on the `decryptOutput, err := client.Decrypt(context.TODO(), decryptInput)` line below

```go
func TestKMSKeyFlow(t *testing.T) {
	// Generate a key
	publicKeyString, keyId := makeKey()

	// Encrypt payload using the public key
        pubKeyInterface, _ := x509.ParsePKIXPublicKey([]byte(publicKeyString))
	publicKey := pubKeyInterface.(*rsa.PublicKey)
	plaintext := []byte("Hello, World!")
	rsaCiphertext, err := crypto.EncryptRSAOAEP(publicKey, plaintext)
	if err != nil {
		t.Errorf("Failed to encrypt plaintext: %v", err)
	}

	// Decrypt the payload
        decryptedMsg := decryptMessage(string(rsaCiphertext), keyId)

	if string(plaintext) != string(decryptedMsg) {
		t.Errorf("Decrypted msg does not match original msg")
	}
}

func makeKey() (string, string) {
	client := kmsClient()
	// Create the RSA_4096 key``
	createKeyInput := &kms.CreateKeyInput{
		KeySpec:     types.KeySpec("RSA_4096"),
		KeyUsage:    types.KeyUsageType("ENCRYPT_DECRYPT"),
		Description: aws.String("RSA_4096 key for encryption and decryption"),
		Tags: []types.Tag{
			{
				TagKey:   aws.String("owner"),
				TagValue: aws.String("oidc-sts"),
			},
		},
	}

	createKeyOutput, err := client.CreateKey(context.TODO(), createKeyInput)
	if err != nil {
		log.Fatalf("Failed to create RSA_4096 key: %s", err)
	}

	getPubKeyInput := &kms.GetPublicKeyInput{
		KeyId: createKeyOutput.KeyMetadata.KeyId,
	}

	keyMaterial, err := client.GetPublicKey(context.TODO(), getPubKeyInput)
	if err != nil {
		log.Fatalf("Failed to get public key for RSA_4096 key: %s", err)
	}

	return string(keyMaterial.PublicKey), *createKeyOutput.KeyMetadata.KeyId
}

func decryptMessage(ciphertext string, keyID string) []byte {
	client := kmsClient()
	decryptInput := &kms.DecryptInput{
		CiphertextBlob: []byte(ciphertext),
		KeyId:          aws.String(keyID),
	}

	decryptOutput, err := client.Decrypt(context.TODO(), decryptInput)
	if err != nil {
		log.Fatalf("Failed to decrypt message: %s", err)				<--- Fails here
	}

	return decryptOutput.Plaintext
}
```

### Environment

```markdown
- OS: macOS 14.6
- LocalStack: 
  LocalStack version: 3.6.1.dev
  LocalStack Docker image sha: cfd66eaca18dc876c097e4f260fdc52120fcd929a0407e370d70aefc24bf6b15
  LocalStack build date: 2024-07-26
  LocalStack build git hash: 428882f0b
```

### Anything else?

I am happy to provide more context, or if there is an obvious section of code, I could look at trying for a PR.

**Comment by sannya-singal** [comment]
Hi @getglad 👋 Thanks for reporting this issue. We already have this item in our backlogs and is noted in the limitations section of our official documentation: https://docs.localstack.cloud/user-guide/aws/kms/. 

If you’re interested in contributing, I’d be happy to help guide you. To start with kindly check the contribution guidelines https://github.com/localstack/localstack/blob/master/docs/CONTRIBUTING.md.

**Comment by getglad** [comment]
@sannya-singal thanks for the response!

> We already have this item in our backlog

Just to be sure I'm following - are you saying it's an improvement you'd accept but isn't scheduled to be worked on?

Having taken a look around I think I see [the order of operations that could be flipped](https://github.com/localstack/localstack/blob/a747f17096c838c1a9f2b2acc0f1785edb07a906/localstack-core/localstack/services/kms/provider.py#L977-L989) but I imagine there will be some learning by doing in the [decrypt](https://github.com/localstack/localstack/blob/a747f17096c838c1a9f2b2acc0f1785edb07a906/localstack-core/localstack/services/kms/provider.py#L995) function attached to the `KmsKey` class.

If there are any good pointers or known gotchas around that flow I'd appreciate it. But I can try to make some time and see what I can do

**Comment by sannya-singal** [comment]
> Just to be sure I'm following - are you saying it's an improvement you'd accept but isn't scheduled to be worked on?

Yes, exactly. 

I noticed that you have opened a PR for its impllementation 🙌, let me have a look at it shortly! Thank you 🚀

**Comment by sannya-singal** [comment]
thanks for fixing this @getglad!

**Comment by dhudec** [comment]
The fix in #11359 hardcodes SHA-256 for the OAEP padding and ignores the `EncryptionAlgorithm` parameter, so this is still broken for `RSAES_OAEP_SHA_1`. Filed #13864 to track.
